cypress/support/pages/signUp.pageObject.js

```diff
@@ -0,0 +1,47 @@
+import PageObject from '../PageObject';
+
+class SignUpPageObject extends PageObject {
+  url = '/#/register';
+
+  get usernameField() {
+    return cy.getByDataQa('username-sign-up');
+  }
+
+  fillUsernameField(username) {
+    this.usernameField.type(username);
+  }
+
+  get emailField() {
+    return cy.getByDataQa('email-sign-up');
+  }
+
+  fillEmailField(email) {
+    this.emailField.type(email);
+  }
+
+  get passwordField() {
+    return cy.getByDataQa('password-sign-up');
+  }
+
+  fillPasswordField(password) {
+    this.passwordField.type(password);
+  }
+
+  get signUpBtn() {
+    return cy.getByDataQa('sign-up-btn');
+  }
+
+  clickOnSignUpBtn() {
+    this.signUpBtn.click();
+  }
+
+  get registrationFailedModal() {
+    return cy.get('.swal-modal');
```

Comment:
For maintainability, consider using a `data-qa` attribute instead of selecting by class name with `cy.get('.swal-modal')`. This will make the tests less brittle in the face of UI changes.